pyproject.toml

```diff
@@ -6,7 +6,7 @@ build-backend = "setuptools.build_meta"
 name = "argilla"
 description = "Open-source tool for exploring, labeling, and monitoring data for NLP projects."
 readme = "README.md"
-requires-python = ">=3.7"  # TODO: upgrade minimum Python version to 3.8, and pin maximum to < 3.12
+requires-python = ">=3.8,<3.12"
```

Comment:
We have to review how we use the `Enum`s classes as in Python 3.11 it has changed and it might require us to get the value explicitly in many places